program/src/processor.rs

```diff
@@ -1415,6 +1415,86 @@ impl Processor {
         Ok(())
     }
 
+    #[inline(never)]
+    /// Create a new OpenOrders PDA then
+    /// Call the init_open_orders instruction in serum dex and add this OpenOrders account to margin account
+    fn create_spot_open_orders(program_id: &Pubkey, accounts: &[AccountInfo]) -> MangoResult {
+        const NUM_FIXED: usize = 8;
+        let accounts = array_ref![accounts, 0, NUM_FIXED];
+        let [
+            mango_group_ai,         // read
+            mango_account_ai,       // write
+            owner_ai,               // read
+            dex_prog_ai,            // read
+            open_orders_ai,         // write
+            spot_market_ai,         // read
+            signer_ai,              // read
+            system_prog_ai,         // read
+        ] = accounts;
+        check!(
+            system_prog_ai.key == &solana_program::system_program::id(),
+            MangoErrorCode::InvalidProgramId
+        )?;
+
+        let mango_group = MangoGroup::load_checked(mango_group_ai, program_id)?;
+        check_eq!(dex_prog_ai.key, &mango_group.dex_program_id, MangoErrorCode::InvalidProgramId)?;
+        check!(&mango_group.signer_key == signer_ai.key, MangoErrorCode::InvalidSignerKey)?;
+
+        let market_index = mango_group
+            .find_spot_market_index(spot_market_ai.key)
+            .ok_or(throw_err!(MangoErrorCode::InvalidMarket))?;
+
+        let mut mango_account =
+            MangoAccount::load_mut_checked(mango_account_ai, program_id, mango_group_ai.key)?;
+        check!(
+            &mango_account.owner == owner_ai.key || &mango_account.delegate == owner_ai.key,
+            MangoErrorCode::InvalidOwner
+        )?;
+        check!(owner_ai.is_signer, MangoErrorCode::InvalidSignerKey)?;
+        check!(!mango_account.is_bankrupt, MangoErrorCode::Bankrupt)?;
+
+        let open_orders_seeds: &[&[u8]] =
+            &[&mango_account_ai.key.as_ref(), &market_index.to_le_bytes(), b"OpenOrders"];
+        seed_and_create_pda(
+            program_id,
+            owner_ai,
+            &Rent::get()?,
```

Comment:
Reading this now: I thought the `Rent` sysvar needs to be passed in? Anchor requires it, but seems unnecessary? That could simplify some calls in voter stake registry etc @microwavedcola1